src/checkpoint/mysql.ts

```diff
@@ -1,19 +1,51 @@
-import mysql from 'mysql';
+import mysql, { PoolConfig, Pool as PoolType, QueryOptions } from 'mysql';
 import Pool from 'mysql/lib/Pool';
 import Connection from 'mysql/lib/Connection';
 import bluebird from 'bluebird';
-import parse from 'connection-string';
+import { ConnectionString } from 'connection-string';
 
-// @ts-ignore
-const config = parse(process.env.DATABASE_URL);
-config.connectionLimit = 1;
-config.multipleStatements = true;
-config.database = config.path[0];
-config.host = config.hosts[0].name;
-config.port = config.hosts[0].port;
-config.connectTimeout = 30000;
-config.charset = 'utf8mb4';
 bluebird.promisifyAll([Pool, Connection]);
-const db = mysql.createPool(config);
 
-export default db;
+/**
+ * Type definition for the promisified Pool type.
+ *
+ * This has to be updated manually with new promisified methods
+ * that users will like to access.
+ */
+export interface AsyncMySqlPool extends PoolType {
+  queryAsync: (options: string | QueryOptions, values?: any) => Promise<any>;
+}
+
+/**
+ * Attempts to connect to the database by the connection string. If no connection string
+ * argument is provided, it tries to use the `DATABASE_URL` environment variable
+ * as connection string.
+ *
+ * This returns a mysql pool connection object.
+ */
+export const createMySqlPool = (connection?: string): AsyncMySqlPool => {
+  if (!connection && !process.env.DATABASE_URL) {
+    throw new Error(
+      'a valid connection string or DATABASE_URL environment variable is required to connect to the database'
+    );
+  }
+
+  const connectionConfig = new ConnectionString((connection || process.env.DATABASE_URL) as string);
+  if (!connectionConfig.hosts || !connectionConfig.path) {
+    throw new Error('invalid mysql connection string provided');
+  }
+
+  const config: PoolConfig = {
+    connectionLimit: 1,
```

Comment:
We will probably need a way to customize the connectionLimit in future if we want support large number of queries, not too important now